template/pages/_document.tsx

```diff
@@ -0,0 +1,79 @@
+import React from 'react';
+import Document, {
+  Html,
+  Head,
+  Main,
+  NextScript,
+  DocumentContext,
+  DocumentProps as NextDocumentProps,
+} from 'next/document';
+import { extractCritical } from 'emotion-server';
+import { EmotionCritical } from 'create-emotion-server';
+
+interface DocumentProps extends NextDocumentProps, EmotionCritical {}
+
+export default class extends Document<DocumentProps> {
+  /**
+   * Manually configuring Emotion for server-side rendering (SSR) enables
+   * the safe use of :nth-child() selectors.
+   */
+  static async getInitialProps({ renderPage, ...ctx }: DocumentContext) {
+    const page = await renderPage();
+    const styles = extractCritical(page.html);
+    return { ...ctx, ...page, ...styles };
+  }
+
+  render() {
+    return (
+      <Html lang={process.env.SITE_LOCALE}>
+        <Head>
```

Comment:
Why are you not using the `meta`  component?

Reply:
I assume you mean the `Meta` React component, not the `meta` HTML element, right? 

The `_document` component is a special one because it is (only) rendered on the server. The `Head` component here is different than the one used in the `Meta` component (which is isomorphic). 

From the [Next.js docs](https://nextjs.org/docs/advanced-features/custom-document):

> The `<Head />` component used here is not the same one from `next/head`. The `<Head />` component used here should only be used for any `<head>` code that is common for all pages. For all other cases, such as `<title>` tags, we recommend using `next/head` in your pages or components.